From 1126865a72a250644ff1807a2f9d0aaec5cc00ef Mon Sep 17 00:00:00 2001
From: Riddhi Aggarwal <quic_riddagga@quicinc.com>
Date: Thu, 4 May 2023 14:24:38 +0530
Subject: [PATCH 218/281] drivers: soc: qcom: Adding rpm log support

Enabling CONFIG_MSM_RPM_LOG by updating Makefile
and Kconfig

Change-Id: I575f65484ad110674249a64df2b8364fce4fbde0
Signed-off-by: Riddhi Aggarwal <quic_riddagga@quicinc.com>
---
 drivers/soc/qcom/Kconfig  | 8 ++++++++
 drivers/soc/qcom/Makefile | 1 +
 2 files changed, 9 insertions(+)

diff --git a/drivers/soc/qcom/Kconfig b/drivers/soc/qcom/Kconfig
index e3ec9ad058c2..df8ad36cc36b 100644
--- a/drivers/soc/qcom/Kconfig
+++ b/drivers/soc/qcom/Kconfig
@@ -254,4 +254,12 @@ config IPQ_REG_UPDATE
 	help
 	 Generic driver to configure required registers
 
+config MSM_RPM_LOG
+	tristate "QTI Resource Power Manager Log Driver"
+	help
+	 This option enables a driver which can read from a circular buffer
+	 of messages produced by the RPM. These messages provide diagnostic
+	 information about RPM operation. The driver outputs the messages
+	 via a debugfs node.
+
 endmenu
diff --git a/drivers/soc/qcom/Makefile b/drivers/soc/qcom/Makefile
index c7ca1a5bde91..8b44c96c29ee 100644
--- a/drivers/soc/qcom/Makefile
+++ b/drivers/soc/qcom/Makefile
@@ -30,3 +30,4 @@ obj-$(CONFIG_QCOM_RPMPD) += rpmpd.o
 obj-$(CONFIG_QCOM_KRYO_L2_ACCESSORS) +=	kryo-l2-accessors.o
 obj-$(CONFIG_QCOM_ICC_BWMON)	+= icc-bwmon.o
 obj-$(CONFIG_IPQ_REG_UPDATE) += ipq_reg_update.o
+obj-$(CONFIG_MSM_RPM_LOG)	+= rpm_log.o
-- 
2.17.1

